Source/DafnyPipeline.Test/ExternalResolverTest.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Xunit;
+using Microsoft.Dafny;
+
+namespace DafnyPipeline.Test;
+
+[Collection("External Resolver plug-in tests")]
+public class ExternalResolverTest {
+  /// <summary>
+  /// This method creates a library and returns the path to that library.
+  /// The library extends an IRewriter so that we can verify that Dafny invokes it if provided in argument.
+  /// </summary>
+  public async Task<string> GetLibrary(string code) {
+    var temp = Path.GetTempFileName();
+    var compilation = CSharpCompilation.Create("tempAssembly");
+    var standardLibraries = new List<string>()
+    {
+      "DafnyPipeline",
+      "System.Runtime",
+      "Boogie.Core"
+    };
+    compilation = compilation.AddReferences(standardLibraries.Select(fileName =>
+        MetadataReference.CreateFromFile(Assembly.Load(fileName).Location)))
+      .AddReferences(
+        MetadataReference.CreateFromFile(typeof(object).GetTypeInfo().Assembly.Location))
+      .WithOptions(
+        new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary)
+      );
+    var syntaxTree = CSharpSyntaxTree.ParseText(code);
+    compilation = compilation.AddSyntaxTrees(syntaxTree);
+    var assemblyPath = $"{temp}.dll";
+    var result = compilation.Emit(assemblyPath);
+
+    var configuration = JsonSerializer.Serialize(
+      new {
+        runtimeOptions = new {
+          tfm = "net6.0",
+          framework = new {
+            name = "Microsoft.NETCore.App",
+            version = "6.0.0",
+            rollForward = "LatestMinor"
+          }
+        }
+      }, new JsonSerializerOptions() { WriteIndented = true });
+    await File.WriteAllTextAsync(temp + ".runtimeconfig.json", configuration + Environment.NewLine);
```

Comment:
Is the `*.runtimeconfig.json` file necessary for dynamically loaded assemblies?

Reply:
I don't know, I just copied-pasted the code of @keyboardDrummer . I removed it and it works, thanks for reporting.

Reply:
Perfect. I guess you've copied this from @keyboardDrummer's code that test's the shutdown of the language server if the parent process quits. The difference to his implementation, is that you're creating a DLL that's loaded by a running process while his implementation is used to spawn a new process. However, I'm not sure if the `*.runtimeconfig.json` file is required if the application is launched with the `dotnet` command; it sure is when calling the executable directly.